src/trace-plugin-loader.js

```diff
@@ -140,49 +109,30 @@ function activate(agent) {
         if (patch.intercept) {
           patch.module = patch.intercept(patch.module, api);
         }
-        patch.active = true;
       });
-      var rootPatch = patchSet.filter(function(patch) { return !patch.name; })[0];
+      var rootPatch = patchSet[''];
       if (rootPatch && rootPatch.intercept) {
         return rootPatch.module;
       } else {
         return null;
       }
     }
 
-    function moduleAlreadyPatched(instrumentation, moduleRoot) {
-      if (!instrumentation.patches[moduleRoot]) {
-        return false;
-      }
-      var modulePatch = instrumentation.patches[moduleRoot];
-      return !!modulePatch && Object.keys(modulePatch).every(function(curr) {
-        return modulePatch[curr].active;
-      }, true);
-    }
-
-    // If this is a reactivation, we may have a cached list of modules from last
-    // time that we need to go and patch pro-actively.
-    for (var moduleName in plugins) {
-      var instrumentation = plugins[moduleName];
-      for (var moduleRoot in instrumentation.patches) {
-        var modulePatch = instrumentation.patches[moduleRoot];
-        if (modulePatch) {
-          loadAndPatch(instrumentation, moduleRoot, null);
-        }
-      }
+    function moduleAlreadyPatched(instrumentation, moduleRoot, version) {
+      return instrumentation.patches[moduleRoot] &&
+        instrumentation.patches[moduleRoot][version];
     }
 
     // Future requires get patched as they get loaded.
     return function Module_load(request, parent, isMain) {
       var instrumentation = plugins[request];
 
-      if (instrumentation &&
-          agent.config().excludedHooks.indexOf(request) === -1) {
```

Comment:
That makes sense, once plugins are included explicitly we won't have need of this configuration any more.